azpubsub/src/main/java/com/microsoft/log4j/DatedRollingFileAppender.java

```diff
@@ -94,16 +105,34 @@ public DatedRollingFileAppender(Layout layout, String filename) throws IOExcepti
         super(layout, filename);
     }
 
+    /**
+     * Get the maximum size that the output file is allowed to reach
+     * before being rolled over to backup files.
+     */
+    public long getMaximumFileSize() {
+        return maxFileSize;
+    }
+
     /**
      * Returns the value of the <b>MaxBackupIndex</b> option.
      */
     public int getMaxBackupIndex() {
         return maxBackupIndex;
     }
 
+    /**
+     * Sets and opens the file where the log output will go. The specified file must be writable.
+     * If there was already an opened file, then the previous file is closed first.
+     *
+     * Overrides setFile in super class FileAppender
+     */
     public synchronized void setFile(String fileName, boolean append, boolean bufferedIO, int bufferSize) throws IOException {
         this.originalFileName = fileName;
-        super.setFile(this.originalFileName + sdf.format(new Date()), append, bufferedIO, bufferSize);
+        String currentFile = this.originalFileName + sdf.format(new Date()) + "." + getSerialNumber();
+        super.setFile(currentFile, append, bufferedIO, bufferSize);
+        // set the qw counter to current/latest file length
+        File file = new File(currentFile);
```

Comment:
forgot about setting the file size if we end up using the exist file to append... we can tweak the existing method to return both serial number & file size